opentutor_classifier/opentutor_classifier/lr_trans_emb_diff/predict.py

```diff
@@ -0,0 +1,155 @@
+#
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+from sklearn import linear_model
+
+from opentutor_classifier import (
+    ARCH_LR_TRANS_EMB_DIFF_CLASSIFIER,
+    AnswerClassifier,
+    AnswerClassifierInput,
+    AnswerClassifierResult,
+    ClassifierConfig,
+    ExpectationClassifierResult,
+    ModelRef,
+    QuestionConfig,
+)
+from opentutor_classifier.dao import find_predicton_config_and_pickle
+from opentutor_classifier.speechact import SpeechActClassifier
+from opentutor_classifier.utils import model_last_updated_at
+from .constants import MODEL_FILE_NAME
+from .dtos import ExpectationToEvaluate, InstanceModels
+from .expectations import LRExpectationClassifier
+from .features import preprocess_sentence
+from opentutor_classifier.sentence_transformer import find_or_load_sentence_transformer
+
+
+def _confidence_score(
+    model: linear_model.LogisticRegression, sentence: List[List[float]]
+) -> float:
+    return model.predict_proba(sentence)[0, 1]
+
+
+ModelAndConfig = Tuple[Dict[int, linear_model.LogisticRegression], QuestionConfig]
+
+
+class LRAnswerClassifier(AnswerClassifier):
+    def __init__(self):
+        self.sentence_transformer = None
+        self._instance_models: Optional[InstanceModels] = None
+        self.speech_act_classifier = SpeechActClassifier()
+        self._model_and_config: ModelAndConfig = None
+        self.exp_ideal_answer = dict()
+
+    def configure(
+        self,
+        config: ClassifierConfig,
+    ) -> AnswerClassifier:
+        self.dao = config.dao
+        self.model_name = config.model_name
+        self.model_roots = config.model_roots
+        self.shared_root = config.shared_root
+        return self
+
+    @property
+    def model_and_config(self) -> ModelAndConfig:
+        if not self._model_and_config:
+            cm = find_predicton_config_and_pickle(
+                ModelRef(
+                    arch=ARCH_LR_TRANS_EMB_DIFF_CLASSIFIER,
+                    lesson=self.model_name,
+                    filename=MODEL_FILE_NAME,
+                ),
+                self.dao,
+            )
+            self._model_and_config = (cm.model, cm.config)
+        return self._model_and_config
+
+    def find_model_for_expectation(
+        self,
+        m_by_e: Dict[int, linear_model.LogisticRegression],
+        expectation: int,
+        return_first_model_if_only_one=False,
+    ) -> linear_model.LogisticRegression:
+        return (
+            m_by_e[0]
+            if expectation >= len(m_by_e) and return_first_model_if_only_one
+            else m_by_e[expectation]
+        )
+
+    def find_sentence_transformer(self):
+        self.sentence_transformer = find_or_load_sentence_transformer(
+            self.shared_root + "/../sentence-transformer"
```

Comment:
use `path.join` and let's get rid of the backstep

Reply:
How can I get rid of backstep?...because shared_root starts one step ahead of the sentence transformer directory.